### ISSUE TITLE
"Could not convert the File Changes Service to a TFS Pending Changes Service. This should never happen!"

### ISSUE THREAD

**Issue by davidjbroadhead** [deadlydog/VS.DiffAllFiles#19]
Getting the error "Could not convert the File Changes Service to a TFS Pending Changes Service" when doing a diff all for a changeset open in the team explorer window but then opening another changeset whilst the files are still iterating in another comparison tool.

**Comment by deadlydog** [comment]
To clarify, you are saying that you experience the error when you start iterating through the changeset files with Diff All Files, but then navigate to another changeset in the Team Explorer window before Diff All Files has finished iterating through all of the files in the first changeset? Is that correct @davidjbroadhead ?

**Comment by davidjbroadhead** [comment]
Yes that's right. I use winmerge and it keeps opening blank comparisons for each file remaining once navigating to another changeset

**Comment by deadlydog** [comment]
Ok, that's a use-case I had never considered or tested. I haven't touched this extension in quite a while, but will keep this in mind the next time I revisit it. It sounds like Visual Studio 2019 will be coming out later this year, so I'll probably need to update the extension to support it, and can try and add this functionality at that time. To be clear, the desired functionality is that is should simply cancel any pending diffs that haven't been opened yet when you navigate away from the changeset in Team Explorer, correct?

I'm also open to Pull Requests if you're inclined to come up with a fix yourself ;)

**Comment by davidjbroadhead** [comment]
Ok, yes I think cancelling it would be right. Thank you!
